baaas-decision-fleet-manager/src/main/java/org/kie/baaas/dfm/app/manager/kafkaservice/KafkaServiceFactory.java

```diff
@@ -0,0 +1,42 @@
+/*
+ * Copyright 2021 Red Hat, Inc. and/or its affiliates.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *   Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ */
+
+package org.kie.baaas.dfm.app.manager.kafkaservice;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+import org.kie.baaas.dfm.app.config.DecisionFleetManagerConfig;
+import org.kie.baaas.dfm.app.manager.KafkaService;
+
+@ApplicationScoped
+public class KafkaServiceFactory {
+    private final DecisionFleetManagerConfig fmConfig;
+
+    @Inject
+    public KafkaServiceFactory(DecisionFleetManagerConfig config) {
+        this.fmConfig = config;
+    }
+
+    public KafkaService createKafkaClient() {
```

Comment:
@myeung18 Take a look at `Producer` declarations in CDI: https://quarkus.io/guides/cdi-reference#simplified-producer-method-declaration

We only need to support one instance of a `KafkaService` per-environment, so we could do something like:

```
class KafkaServiceProducer {

  @ConfigProperty(name = "baaas.dfm.kafka.service.type")
  String serviceType;

  @ApplicationScoped
  KafkaService produceKafkaService() {
        KafkaService kc = new UndefinedKafkaService();
        if (KafkaServiceType.OPERATE_FIRST.equalValue(fmConfig.getKafkaServiceType())) {
            kc = new OperateFirstKafkaService();
        } else if (KafkaServiceType.MANAGED_KAFKA.equalValue(fmConfig.getKafkaServiceType())) {
            kc = new ManagedKafkaService();
        }
        return kc;
  }
}
```

This would mean that the `KafkaService` instance can then be injected into the `DecisionLifecycleOrchestrator` rather than the `KafkaServiceFactory`.

Reply:
yes, agreed.